### ISSUE TITLE
State Trie Migration

### ISSUE THREAD

**Issue by ggwpez** [paritytech/polkadot-sdk#4174]
Need to track where we are and what is missing cc @cheme @PierreBesson.  
Three check marks: migration deployed, RPC reports done, migration removed from the runtime.

- [ ] Rococo [x], [x], [ ]
  - [ ] Asset-Hub [x], [x], [ ]
  - [x] ~~Bridge-Hub [ ], [ ], [ ], [ ]~~
  - [x] ~~Contracts [ ], [ ], [ ], [ ]~~
  - [x] ~~Coretime [ ], [ ], [ ], [ ]~~
  - [x] ~~People [ ], [ ], [ ], [ ]~~
- [ ] Westend [x], [x], [ ]
  - [ ] Asset-Hub [x], [ ], [ ]
  - [x] ~~Bridge-Hub [ ], [ ], [ ], [ ]~~
  - [ ] Collectives [x], [ ], [ ]
  - [x] ~~Coretime [ ], [ ], [ ], [ ]~~
  - [x] ~~Glutton [ ], [ ], [ ], [ ]~~ 
  - [x] ~~People [ ], [ ], [ ], [ ]~~

**Comment by ggwpez** [comment]
@cheme i checked all the runtimes and i think i got all the missing ones listed here.  
Could you please add the migration to the missing ones? Seems like most new chains have V1 since genesis, so its not that many.

**Comment by cheme** [comment]
Thanks, 
> Could you please add the migration to the missing ones?

Will do tomorrow (most likely will be the same change as the one you adapt in https://github.com/paritytech/polkadot-sdk/commit/cecb8f4e343d6fd3f9d2bde53969fff920cf0b33 , but I will reread the pallet logic first , just in case.

> Seems like most new chains have V1 since genesis, so its not that many.

yes new chain should have started v1 (I am a bit surprise by asset_hub (we migrate at least two statemine/statemint envt), but it can be done any number of time anyway).

Did quickly check and see same thing.
Note that the three first check marks make sense to me, but not the fourth one: the state_version is switch in the runtime at the same time as the runtime with the migration is deployed (the migration only works if the state_version is 1 in the runtime).

**Comment by ggwpez** [comment]
> Did quickly check and see same thing.
> Note that the three first check marks make sense to me, but not the fourth one: the state_version is switch in the runtime at the same time as the runtime with the migration is deployed (the migration only works if the state_version is 1 in the runtime).

Okay, will remove 👍

**Comment by ggwpez** [comment]
Enactment for for AH Westend and Collectives Westend should happen with 1.12 upgrade as it was added in https://github.com/paritytech/polkadot-sdk/pull/4185.

**Comment by ggwpez** [comment]
I think AH Rococo completed and we can remove the pallet, or @cheme? It returns this:
```pre
{
  progressTop: Complete
  progressChild: ToStart
  size_: 841,970
  topItems: 2,976
  childItems: 0
}
```
We dont have a node with RPC enabled, but given that the last `Migrated` events was [1.5 years back](https://assethub-rococo.subscan.io/event?page=1&time_dimension=date&module=statetriemigration), i think its safe to remove, or?

**Comment by cheme** [comment]
yes it was when we did test the migration with Pierre, they are finished.
